blocks/library/tweet/index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Internal dependencies
+ */
+import { registerBlock, query } from '../../api';
+import Sandbox from '../../../components/sandbox';
+import Button from '../../../components/button';
+import Placeholder from '../../../components/placeholder';
+
+const { prop } = query;
+
+registerBlock( 'core/tweet', {
+	title: wp.i18n.__( 'Tweet' ),
+	icon: 'twitter',
+
+	category: 'social',
+
+	attributes: {
+		url: prop( '*', 'innerHTML' ), // our html is just a div with the url in, for WP's oembed to process
```

Comment:
For a case like this, `attributes` can be defined as a function which receives the raw content of the block as a string, which seems to be exactly what you want:

```js
attributes( url ) {
	return { url };
}
```

https://github.com/WordPress/gutenberg/tree/master/blocks#wpblocksregisterblock-slug-string-settings-object-